src/transport.ts

```diff
@@ -58,52 +58,39 @@ export class WebRTCTransport implements Transport, Initializable {
       throw inappropriateMultiaddr("we need to have the remote's PeerId");
     }
 
-    let peerConnection = new RTCPeerConnection();
+    let certificate = await RTCPeerConnection.generateCertificate({
+      name: 'ECDSA',
+      namedCurve: 'P-256',
+    } as any);
+    let peerConnection = new RTCPeerConnection({ certificates: [certificate] });
 
     // create data channel
+    let dataChannelOpenPromise = defer();
     let handshakeDataChannel = peerConnection.createDataChannel('data', { negotiated: true, id: 1 });
-    //
+    handshakeDataChannel.onopen = (_) => dataChannelOpenPromise.resolve();
+    handshakeDataChannel.onerror = (ev: Event) => {
+      log.error('Error opening a data channel for handshaking: %s', ev.toString());
+      dataChannelOpenPromise.reject(dataChannelError('data', `error opening datachannel: ${ev.toString()}`));
+    };
+    setTimeout(() => {
+      log.error('Data channel never opened. State was: %s', handshakeDataChannel.readyState.toString());
+      dataChannelOpenPromise.reject(dataChannelError('data', `data channel was never opened: state: ${handshakeDataChannel.readyState}`));
+    }, HANDSHAKE_TIMEOUT_MS);
+
     // create offer sdp
     let offerSdp = await peerConnection.createOffer();
-    //
-    //
     // generate random string for ufrag
-    let ufrag = genUuid();
-
-    //
+    let ufrag = genUuid().replaceAll('-', '');
```

Comment:
is there now ay to ge the uuid without having to replace chars?

Reply:
We don't really need a uuid, just a random string. Removing the `-` character is a webrtc requirement where the ufrag and password cannot have that character.